src/components/EditableTagField/index.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useEffect } from "react";
+import styled from "@emotion/styled";
+import { InstanceTag, ParameterInput } from "gql/generated/types";
+import { convertArrayToObject } from "utils/array";
+import { TagRow } from "./TagRow";
+
+type Tag = InstanceTag | ParameterInput;
+type EditableTagFieldProps = {
+  onChange: (data: Tag[]) => void;
+  inputTags: Tag[];
+  visible?: boolean;
+};
+
+export const EditableTagField: React.FC<EditableTagFieldProps> = ({
+  onChange,
+  inputTags,
+  visible,
+}) => {
+  const [visibleTags, setVisibleTags] = useState(inputTags);
```

Comment:
Can we take in setVisibleTags as a prop so that the page using this component can access all the tags and do something with it?

Reply:
I dont think that is a good idea as that would imply an outside component would control the internal state of this component. The component already provides an interface to display and make changes to the tags rendered with in it.

tags can be passed in with the `inputTags` prop and any changes to the state of the tags will be returned with the `onChange` callback prop.